Invalid XML with PHP anonymous class

Hello :wave: 

I would like to know if there's an update regarding the [pre-parsing sanitization](https://github.com/vim-vdebug/vdebug/issues/60#issuecomment-175763879) @joonty ?

I've run into an issue similar to #60 today, the parsed xml is:

```xml
<response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="context_get" transaction_id="17" context="0">
    <property name="$deferred" fullname="$deferred" type="object" classname="Amp\Deferred" children="1" numchildren="2" page="0" pagesize="32">
        <property name="resolver" fullname="$deferred-&gt;resolver" facet="private" type="object" classname="class@anonymous&#0;/app/vendor/amphp/amp/lib/Deferred.php0x7fa0c40e72d3" children="1" numchildren="4">
        </property>
        <property name="promise" fullname="$deferred-&gt;promise" facet="private" type="object" classname="Amp\Internal\PrivatePromise" children="1" numchildren="1">
        </property>
    </property>
</response>
```

which is invalid (because of "invalid character number" `&#0;`).

The full vdebug error trace is:

```
- [ERROR] {Tue 17 2018 13:05:36} An error occured: <class 'xml.etree.ElementTree.ParseError'>
Traceback (most recent call last):
  File "/home/gildas/.vim/plugged/vdebug/pythonx/vdebug/event.py", line 524, in dispatch_event
    Dispatcher.events[name](self.__session_handler).run(*args)
  File "/home/gildas/.vim/plugged/vdebug/pythonx/vdebug/event.py", line 439, in run
    self.ui.windows.watch().accept_renderer(rend)
  File "/home/gildas/.vim/plugged/vdebug/pythonx/vdebug/ui/vimui.py", line 513, in accept_renderer
    self.write(renderer.render())
  File "/home/gildas/.vim/plugged/vdebug/pythonx/vdebug/ui/vimui.py", line 736, in render
    properties = self.response.get_context()
  File "/home/gildas/.vim/plugged/vdebug/pythonx/vdebug/dbgp.py", line 132, in get_context
    for c in list(self.as_xml()):
  File "<string>", line None
xml.etree.ElementTree.ParseError: reference to invalid character number: line 2, column 397
``` 

And here's the PHP version I'm using:
```
PHP 7.2.3 (cli) (built: Mar  5 2018 21:28:31) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Xdebug v2.6.0, Copyright (c) 2002-2018, by Derick Rethans
```

I'm using version 2.0.0 of Vdebug.

Sorry if this has been issued before, I haven't found any reference to it.

Thanks in advance.

--- a/python3/vdebug/connection.py
+++ b/python3/vdebug/connection.py
@@ -81,7 +81,7 @@ class ConnectionHandler:
 
         cmd -- command to send
         """
-        #self.sock.send(cmd + '\0')
+        # self.sock.send(cmd + '\0')
         MSGLEN = len(cmd)
         totalsent = 0
         while totalsent < MSGLEN:

--- a/python3/vdebug/session.py
+++ b/python3/vdebug/session.py
@@ -37,13 +37,14 @@ class SessionHandler:
             print("Waiting for a connection: none found so far")
         elif self.listener and self.listener.is_ready():
             print("Found connection, starting debugger")
+            log.Log("Got connection, starting", log.Logger.DEBUG)
             self.__new_session()
         else:
             self.start_listener()
 
     def start_listener(self):
         self.listener = listener.Listener.create()
-        print("Vdebug will wait for a connection in the background")
+        print("Vdebug will wait for a connection")
         util.Environment.reload()
         if self.is_open():
             self.ui().set_status("listening")
@@ -103,6 +104,7 @@ class SessionHandler:
         try:
             if self.listener.is_ready():
                 print("Found connection, starting debugger")
+                log.Log("Got connection, starting", log.Logger.DEBUG)
                 self.__new_session()
                 return True
             return False
@@ -206,7 +208,9 @@ class Session:
             self.__ui.set_conn_details(addr[0], addr[1])
 
             self.__collect_context_names()
-            self.__set_features()
+            self.__check_features()  # only for debugging at the moment
+            self.__set_default_features()  # features we try by default
+            self.__set_features()  # user defined features
             self.__initialize_breakpoints()
 
             if opts.Options.get('break_on_open', int) == 1:
@@ -228,6 +232,64 @@ class Session:
             self.__api.detach()
 
         self.close_connection(False)
+
+    def __check_features(self):
+        must_features = [
+            'language_supports_threads',
+            'language_name',
+            'language_version',
+            'encoding',  # has set
+            'protocol_version',
+            'supports_async',
+            'data_encoding',
+            'breakpoint_languages',
+            'breakpoint_types',
+            'resolved_breakpoints',
+            'multiple_sessions',  # has set
+            'max_children',  # has set
+            'max_data',  # has set
+            'max_depth',  # has set
+            'extended_properties',  # has set
+        ]
+        maybe_features = [
+            'supported_encodings',
+            'supports_postmortem',
+            'show_hidden',  # has set
+            'notify_ok',  # has set
+        ]
+        for feature in must_features:
+            try:
+                feature_value = self.__api.feature_get(feature)
+                log.Log(
+                    "Must Feature: %s = %s" % (feature, str(feature_value)),
+                    log.Logger.DEBUG
+                )
+            except dbgp.DBGPError:
+                error_str = "Failed to get feature %s" % feature
+                log.Log(error_str, log.Logger.DEBUG)
+
+        for feature in maybe_features:
+            try:
+                feature_value = self.__api.feature_get(feature)
+                log.Log(
+                    "Maybe Feature: %s = %s" % (feature, str(feature_value)),
+                    log.Logger.DEBUG
+                )
+            except dbgp.DBGPError:
+                error_str = "Failed to get feature %s" % feature
+                log.Log(error_str, log.Logger.DEBUG)
+
+    def __set_default_features(self):
+        features = {
+            'multiple_sessions': 0,  # explicitly disable multiple sessions atm
+            'extended_properties': 1,
+        }
+        for name, value in features.items():
+            try:
+                self.__api.feature_set(name, value)
+            except dbgp.DBGPError as e:
+                error_str = "Failed to set feature %s: %s" % (name, e.args[0])
+                log.Log(error_str, log.Logger.DEBUG)
 
     def __set_features(self):
         """Evaluate vim dictionary of features and pass to debugger.

